src/pickup.cpp

```diff
@@ -169,6 +182,49 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
             veh_tool( "hotplate" );
             return DONE;
 
+        case USE_WASHMACHINE: {
+            bool detergent_is_enough = g->u.crafting_inventory().has_charges( "detergent", 5 );
+            static const std::string filthy( "FILTHY" );
+            bool filthy_items = std::any_of( veh->get_items( washing_machine_part ).begin(),
+                                             veh->get_items( washing_machine_part ).end(),
```

Comment:
Though the function doesn't look expensive, it would be safer to stash the result of `veh->get_items( washing_machine_part ) ` in a temp variable: `const auto items = veh->get_items( washing_machine_part )`.